ir/src/main/scala/org/scalajs/core/ir/Utils.scala

```diff
@@ -75,12 +75,13 @@ object Utils {
     sb.toString
   }
 
-  def printEscapeJS(str: String, out: java.lang.Appendable): Unit = {
+  def printEscapeJS(str: String, out: java.lang.Appendable): Int = {
```

Comment:
I like your solution a lot, it's a lot DRYer than I had thought possible.

However you'll have some difficulty with our binary compat checker, because this method is not binary-compatible with its previous incarnation. You can run it locally with
```
> ir/mimaReportBinaryIssues
```
I think we'll just have to break bin-compat of the tools, here. You can whitelist this method in `project/BinaryIncompatibilities.scala`. See for example [here](https://github.com/scala-js/scala-js/blob/v0.6.15/project/BinaryIncompatibilities.scala#L5-L18) how we do it, and the formatting/comments we use.